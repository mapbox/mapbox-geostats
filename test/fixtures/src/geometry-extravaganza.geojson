{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {
        "foo": "bar"
      },
      "geometry": {
        "type": "Point",
        "coordinates": [125.6, 10.1]
      }
    },
    {
      "type": "Feature",
      "properties": {
        "foo": "bar"
      },
      "geometry": {
        "type": "MultiPoint",
        "coordinates": [[0, 0], [1, 1]]
      }
    },
    {
      "type": "Feature",
      "properties": {
        "foo": "bar"
      },
      "geometry": {
        "type": "MultiLineString",
        "coordinates": [
          [[0, 0], [1, 1], [2, 2]],
          [[10, 10], [11, 11], [12, 12]]
        ]
      }
    },
    {
      "type": "Feature",
      "properties": {
        "bar": 1
      },
      "geometry": {
        "type": "MultiPolygon",
        "coordinates": [
          [
            [
              [101.2, 1.2], [101.8, 1.2], [101.8, 1.8], [101.2, 1.8], [101.2, 1.2]
            ],
            [
              [101.2, 1.2], [101.3, 1.2], [101.3, 1.3], [101.2, 1.3], [101.2, 1.2]
            ],
            [
              [101.6, 1.4], [101.7, 1.4], [101.7, 1.5], [101.6, 1.5], [101.6, 1.4]
            ],
            [
              [101.5, 1.6], [101.6, 1.6], [101.6, 1.7], [101.5, 1.7], [101.5, 1.6]
            ]
          ],
          [
            [
              [100.0, 0.0], [101.0, 0.0], [101.0, 1.0], [100.0, 1.0], [100.0, 0.0]
            ],
            [
              [100.35, 0.35], [100.65, 0.35], [100.65, 0.65], [100.35, 0.65], [100.35, 0.35]
            ]
          ]
        ]
      }
    },
    {
      "type": "Feature",
      "properties": {
        "doodoo": "doo"
      },
      "geometry": {
        "type": "GeometryCollection",
        "geometries": [
          {
            "type": "Point",
            "coordinates": [100.0, 0.0]
          },
          {
            "type": "LineString",
            "coordinates": [[101.0, 0.0], [102.0, 1.0]]
          }
        ]
      }
    },
    {
      "type": "Feature",
      "properties": {
        "baz": true
      },
      "geometry": {
        "type": "Point",
        "coordinates": [100.6, 4.1]
      }
    }
  ]
}
