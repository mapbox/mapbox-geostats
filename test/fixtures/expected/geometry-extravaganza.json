{
  "layerCount": 1,
  "layers": [
    {
      "layer": "geometry-extravaganza",
      "count": 6,
      "attributeCount": 4,
      "attributes": [
        {
          "values": [
            "bar",
            null
          ],
          "count": 2,
          "attribute": "foo",
          "type": "string"
        },
        {
          "values": [
            1,
            null
          ],
          "count": 2,
          "attribute": "bar",
          "type": "number",
          "min": 1,
          "max": 1
        },
        {
          "values": [
            "doo",
            null
          ],
          "count": 2,
          "attribute": "doodoo",
          "type": "string"
        },
        {
          "values": [
            true
          ],
          "count": 1,
          "attribute": "baz",
          "type": "boolean"
        }
      ],
      "geometry": "Point"
    }
  ]
}
